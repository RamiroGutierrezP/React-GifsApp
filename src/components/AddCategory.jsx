import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ( {onNewCategory} ) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (event) => {
        setInputValue(event.target.value)
    }
    
    const handleSubmit = (event) => {
      event.preventDefault();

      const trimmedInputValue = inputValue.trim();

      if (trimmedInputValue.length < 1) return;
      onNewCategory(trimmedInputValue);

      setInputValue('');
    }

    return (
        <>
          <form onSubmit={handleSubmit}>
            <input 
              type="text"
              placeholder="Buscar gifs"
              value={inputValue}
              onChange={ handleInputChange }
            />
          </form>
        </>
    )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
