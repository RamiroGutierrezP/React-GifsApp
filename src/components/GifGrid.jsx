
import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ( { category } ) => {

    const { images, isLoading } = useFetchGifs(category);
        
    return (
        <>
            <h3>{ category }</h3>
            <hr />
            { isLoading && <p>Loading...</p> }
            <div className="card-grid">
                {
                    images.map( ( img ) => (
                        <GifItem
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            </div>
        </> 
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

