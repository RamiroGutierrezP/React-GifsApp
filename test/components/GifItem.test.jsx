/* eslint-disable no-undef */

import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('<GifItem /> test', () => { 

    const title = 'Independiente'
    const url = 'https://infiernorojo.com/cai.jpg'

    test('should match with the snapshot', () => {

        const { container } = render(<GifItem title={ title } url={ url } />)
        expect( container ).toMatchSnapshot()

    })
    test('should show the URL and ALT indicated', () => {  

        render(<GifItem title={ title } url={ url } />)
        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )

    })

})
