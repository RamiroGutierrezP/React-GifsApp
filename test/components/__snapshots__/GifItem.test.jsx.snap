// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> test should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Independiente"
      src="https://infiernorojo.com/cai.jpg"
    />
  </div>
</div>
`;
